nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -113,13 +113,21 @@ def _get_spheres_rows(seeds, target_affine, target_shape, radius, allow_overlap)
     '''
     seeds = list(seeds)
 
-    # Compute world coordinates of all voxels. The mask will be applied at 
-    # the last step.
-    brain_coords = list(np.ndindex(*target_shape))
+    # Compute world coordinates of the mask and of all coordinates
+    mask = _safe_get_data(mask_img)
+    
+    mask_coords = list(zip(*np.where(mask == 0)))
+    mask_coords = np.asarray(list(zip(*mask_coords)))
+    if not np.all(mask == 1):
```

Comment:
Need to use the data which is NOT in the mask. Basically a reverse of the forward function. 

The `if not np.all(mask==1)`  is unfortunately needed for masks that include the whole brain / image. However, everything else will work with empty lists.